clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -204,16 +202,14 @@ public static byte[] compress(byte[] raw, CompressionType compressionType) throw
     public static ByteBuffer decompress(byte[] metrics, CompressionType compressionType) {
         ByteBuffer data = ByteBuffer.wrap(metrics);
         try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create());
-            ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-
+             ByteBufferOutputStream out = new ByteBufferOutputStream(512)) {
```

Comment:
One tiny, tiny comment. The indentation of this line is out by 1 space. Apart from that, lgtm.

Reply:
Done. Thanks for the reminder.